common/unsafe/src/main/java/org/apache/spark/unsafe/types/UTF8String.java

```diff
@@ -1390,8 +1497,13 @@ public UTF8String copy() {
 
   @Override
   public int compareTo(@Nonnull final UTF8String other) {
-    return ByteArray.compareBinary(
-        base, offset, numBytes, other.base, other.offset, other.numBytes);
+    if (comparatorId == 0) {
```

Comment:
what if the `other` has nonzero comparatorId, should we do something like `-1 * other.compareTo(this)`?